src/devices/machine/idehd.cpp

```diff
@@ -379,22 +379,22 @@ void ata_mass_storage_device::security_error()
 
 attotime ata_mass_storage_device::seek_time()
 {
-	int sectors_per_cylinder =  m_num_heads * m_num_sectors;
+	uint32_t sectors_per_cylinder =  m_num_heads * m_num_sectors;
 
 	if (sectors_per_cylinder == 0 || m_num_cylinders == 0)
 		return attotime::zero;
 
-	int new_lba = lba_address();
-	int old_cylinder = m_cur_lba / sectors_per_cylinder;
-	int new_cylinder = new_lba / sectors_per_cylinder;
-	int diff = abs(old_cylinder - new_cylinder);
+	uint32_t new_lba = lba_address();
+	uint16_t old_cylinder = m_cur_lba / sectors_per_cylinder;
+	uint16_t new_cylinder = new_lba / sectors_per_cylinder;
+	uint16_t diff = abs(old_cylinder - new_cylinder);
```

Comment:
Using `abs` on the result of an unsigned subtraction isn't going to end well.  The proper way to make it work is to defensively cast up the results of the divisions to a signed type.

Reply:
It's pointless as we only support 28 bit LBA anyway. So the lba address can't cause a problem for unsigned maths.

Truncating the cylinder to 16 bit is a huge problem.

Reply:
@smf- I mean, the cylinder counts were only 16 bits in the class itself anyway. And the abs was already there before this.

Reply:
@MoochMcGee "I mean, the cylinder counts were only 16 bits in the class itself anyway. And the abs was already there before this."

What do you mean? They were int before you changed them to uint16_t, 16 bits isn't enough.

	int new_lba = lba_address();
	int old_cylinder = m_cur_lba / sectors_per_cylinder;
	int new_cylinder = new_lba / sectors_per_cylinder;
	int diff = abs(old_cylinder - new_cylinder);

Truncating them to 16 bits is just going to turn it into a random number generator, the only difference is they are now only ever 16 bits. So the time will be shorter, which will make you think you're fixing something.